### PR TITLE
Add PIC16F1619 devices to MCU_MicrochipPIC16

### PR DESCRIPTION
**Pull request by TrashHammer**
I updated the MCU_MicrochipPIC16 symbol library to include the PIC16F1619 devices.

Data sheet link:
http://ww1.microchip.com/downloads/en/DeviceDoc/40001770D.pdf

Images of both parts pinout configuration:
Pic16f1829 20pin-2
Pic16f1829 20pin-1
PIC16F1619 pinout

QFN, UQFN version
image

PDIP, SOIC, SSOP
image

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [ ] Give a reason behind any intentional library convention rule violation.

### CODE REVIEW

**Comment by antoniovazquezblanco** [comment]
- [ ] Please, attach screenshots of the devices you have created, not of the datasheet.
- [ ] Please, wildcard the temperature specifier in device names, e.g.: `PIC16F1619-IML` -> `PIC16F1619-xML`

Thanks!

**Comment by myfreescalewebpage** [comment]
@TrashHammer please can you first fix the comment of @antoniovazquezblanco before we start the review ? Also, the device should be atomic, that means, one symbol per package type.

**Comment by TrashHammer** [comment]
> @TrashHammer please can you first fix the comment of @antoniovazquezblanco before we start the review ? Also, the device should be atomic, that means, one symbol per package type.

The parts are now atomic

**Comment by TrashHammer** [comment]
> @TrashHammer please can you first fix the comment of @antoniovazquezblanco before we start the review ? Also, the device should be atomic, that means, one symbol per package type.

**Comment by TrashHammer** [comment]
> @TrashHammer please can you first fix the comment of @antoniovazquezblanco before we start the review ? Also, the device should be atomic, that means, one symbol per package type.

**Comment by TrashHammer** [comment]
I have made the changes to the part numbers to remove the temperature indication
I have made the devices attomic
I have added an additional part to cover all available footprints

**Comment by myfreescalewebpage** [comment]
Thanks, one of us will make the review soon.

File: MCU_Microchip_PIC16.lib [hunk 1]
```diff
@@ -1,6 +1,23 @@
 EESchema-LIBRARY Version 2.4
 #encoding utf-8
 #
+# ExamplePN
```
Comment (antoniovazquezblanco) [inline MCU_Microchip_PIC16.lib]: You have added a ExamplePN device. I guess this was accidental. You should delete it

**Comment by TrashHammer** [comment]
Yes, that was not on purpose
Will delete.

**Comment by TrashHammer** [comment]
Any update on getting this review completed?

**Comment by antoniovazquezblanco** [comment]
You have added new devices that should have a wildcard in the temperature identifier:
`PIC16F18855-MV` -> `PIC16F18855-xMV`
`PIC16F18855-SO` -> `PIC16F18855-xSO`

Please, beware that we do perform reviews on our free time and sometimes this takes a while to get done.

Thanks!

**Comment by TrashHammer** [comment]
The files have been updated as requested.  I do understand the nature of the time involved.  It had been a time long enough that I was questioning if I did not submit the update for review.  No offense was intended.

**Comment by myfreescalewebpage** [comment]
@TrashHammer we have lot of pending PR at this time, it's normal to not been reviewed immediately. It's summer holidays also, which does not help...
Thanks,
Joel

**Comment by myfreescalewebpage** [comment]
@TrashHammer I can review the PR, can you solve the branch conflict please ? Joel

**Comment by TrashHammer** [comment]
This should be good now.

**Comment by TrashHammer** [comment]
This should be complete.

**Comment by myfreescalewebpage** [comment]
No:

Capture

**Comment by TrashHammer** [comment]
I can not figure out how to resolve the conflicts.

**Comment by TrashHammer** [comment]
I think it is updated now.  Sorry for the delay.

**Comment by myfreescalewebpage** [comment]
No worries. However, Git has break the lib during the merge:

- dcm file line 337 there is a missing "#"
- lib file line 1508 there is a missing "ENDDRAW ENDDEF"

KiCad display an error opening the lib

Joel

**Comment by TrashHammer** [comment]
Fixed the file and tested with Kicad.  I have committed and pushed my changes to MCHP

**Comment by myfreescalewebpage** [comment]
Hi @TrashHammer , thanks for the update, here is a partial review for all devices you have submitted:

- [ ] Pin length should be 100mil when the number of pins is < 100
- [ ] Description can be enhanced to described the MCU and should be terminated with the complete package name, see for example "PIC16F1786-xML" in the library
- [ ] Pin offset should be 40mil (see travis warnings)
- [ ] Port should be grouped, for example all pin RCx on the same side
- [ ] Some symbols you added are very large for no reason.

I know you have based some of your devices from existing symbol, but they are not all properly done, we try to make the library better when new symbols are added.

Due to PIC16F18854-SO name modification the pull request can be merged only at the next major release (v6) so we have time to do the wanted modifications here.

Will do a complete review after the above change.

Cheers,
Joel

**Comment by TrashHammer** [comment]
I have made and committed the changes to the parts.

**Comment by TrashHammer** [comment]
@myfreescalewebpage Joel
I am making an additional part for a separate pull request.  I pulled the latest master and created a PIC32MM branch.  I have done my best to make the symbol meets the requirements.  Do I now create a new pull request for the new branch as a separate action?  I was not clear and want to make sure already "ready for review" changes are not held back.

**Comment by myfreescalewebpage** [comment]
> Do I now create a new pull request for the new branch as a separate action

Yes.

I personal create one branch on my fork for each new device I want to add, or family of devices. After pushing the branch I go to my github and create a pull request. If I want to add another device to the same pull request, or to make modifications, I just go on my branch and push again, this automatically update the pull request. If I have another pull request to create, I create another branch. Finally, when a pull request has been merged, I delete my branch on my github.

**Comment by TrashHammer** [comment]
Joel,

Thank you, I am starting to like the git process, but it is taking me time to fully understand how to best use it.

Matt

**Comment by myfreescalewebpage** [comment]
Complete review:

PIC16F1619-xGZ:
- [ ] Name of pin 1 should be `RA3/~MCLR~/VPP`
- [ ] Can you move RCx pins so RC0 is at the top and RC7 at the bottom ?
- [ ] Pin 21 (expose pad) is missing, it should be added just beside VSS, name "EP" and type Passive

PIC16F1619-xML:
- [ ] Name of pin 1 should be `RA3/~MCLR~/VPP`
- [ ] Can you move RCx pins so RC0 is at the top and RC7 at the bottom ?
- [ ] Pin 21 (expose pad) is missing, it should be added just beside VSS, name "EP" and type Passive
- [ ] The package has EP too large, the recommend land pattern specify max 2.5x2.5mm, the footprint should be `Package_DFN_QFN:QFN-20-1EP_4x4mm_P0.5mm_EP2.5x2.5mm`
- [ ] Footprint filter should be `QFN*1EP*4x4mm*P0.5mm*`

PIC16F1619-xP:
- [ ] Name of pin 1 should be `RA3/~MCLR~/VPP`
- [ ] Can you move RCx pins so RC0 is at the top and RC7 at the bottom ?
- [ ] The pinout is copied from previous device, but the PDIP package has a complete different pinout. To be reviewed.
- [ ] The footprint should be `Package_DIP:DIP-20_W7.62mm`

PIC16F1619-xSO:
- [ ] Name of pin 1 should be `RA3/~MCLR~/VPP`
- [ ] Can you move RCx pins so RC0 is at the top and RC7 at the bottom ?
- [ ] The pinout is copied from previous device, but the SOIC package has a complete different pinout. To be reviewed.

PIC16F1619-xSS:
- [ ] Name of pin 1 should be `RA3/~MCLR~/VPP`
- [ ] Can you move RCx pins so RC0 is at the top and RC7 at the bottom ?
- [ ] The pinout is copied from previous device, but the SSOP package has a complete different pinout. To be reviewed.

PIC16F18854-xSO:
- [ ] Pin 19 is missing, it should be stacked with pin 8

PIC16F18854-xMV:
- [ ] Footprint filter should be `UQFN*1EP*4x4mm*P0.4mm*`
- [ ] Pin 29 (expose pad) is missing, it should be added stacked with VSS pins

PIC16F18855-xSO:
- [ ] Should be an alias of PIC16F18854-xSO

**Comment by TrashHammer** [comment]
I made the requested changes.  However, can you explain the proper way to handle multiple power input pins like PIC16F18855?  I updated it until I cleared all the errors from the automatic check, but want to know the accepted and proper practice.

**Comment by TrashHammer** [comment]
I get an error on the travis-ci, but it looks like stacking the VSS and VDD pins it the way to build the symbol based on other parts in the library.  Let me know if I am wrong and need to revert to the prior commit.

**Comment by myfreescalewebpage** [comment]
Last comments:

PIC16F1619-xGZ:
- [ ] You can easily reduce the height of the symbol by 200mil (remove 100mil at the top and remove 100mil between RA and RB ports)

PIC16F1619-xML:
- [ ] You can easily reduce the height of the symbol by 200mil (remove 100mil at the top and remove 100mil between RA and RB ports)

PIC16F1619-xP:
- [ ] You can easily reduce the height of the symbol by 200mil (remove 100mil at the top and remove 100mil between RA and RB ports)

PIC16F1619-xSO:
- [ ] You can easily reduce the height of the symbol by 200mil (remove 100mil at the top and remove 100mil between RA and RB ports)

PIC16F1619-xSS:
- [ ] You can easily reduce the height of the symbol by 200mil (remove 100mil at the top and remove 100mil between RA and RB ports)

PIC16F18854-xSO:
- [ ] You can easily reduce the height of the symbol by 100mil (remove 100mil between RA and RB ports)
Don't worry about the travis error about VSS, the current commit is correct.

PIC16F18855-xMV:
- [ ] You can easily reduce the height of the symbol by 100mil (remove 100mil between RA and RB ports)
Don't worry about the travis error about VSS, the current commit is correct.

Joel

**Comment by TrashHammer** [comment]
I have completed the request and have pushed the changes.

**Comment by myfreescalewebpage** [comment]
Thanks. For the 3 last devices, you have reduced the height too much. Please keep a small space between RA and RB port. The device will not be perfectly centered but it's ok.

**Comment by TrashHammer** [comment]
No problem, I was choosing between the less of two evils so to speak.  I will update it now.

**Comment by TrashHammer** [comment]
I have updated and pushed the changes as requested.

**Comment by myfreescalewebpage** [comment]
Thanks, no more comments on the symbols.
The pull request modify an existing device (PIC16F18854-SO) and thus it will only be possible to merge with the next major version.
Will now wait for v6.
Joel

**Comment by TrashHammer** [comment]
Thanks, I really apprising all your help and patience

Matt

**Comment by myfreescalewebpage** [comment]
Closing/opening to refresh the Travis test.

**Comment by TrashHammer** [comment]
YEAH!!!  That's Awesome.  Thanks again for the help!

**Pull request merged**